lib/generic-pool.js

```diff
@@ -364,25 +364,75 @@ Pool.prototype._ensureMinimum = function _ensureMinimum () {
   }
 }
 
+Pool.prototype._isFull = function() {
+  return this._count >= this._factory.max && this._availableObjects.length === 0
+}
+
 /**
  * Request a new client. The callback will be called,
- * when a new client will be availabe, passing the client to it.
+ * when a new client will be available, passing the client to it.
  *
  * @param {Function} callback
  *   Callback function to be called after the acquire is successful.
  *   The function will receive the acquired item as the first parameter.
  *
  * @param {Number} priority
- *   Optional.  Integer between 0 and (priorityRange - 1).  Specifies the priority
- *   of the caller if there are no available resources.  Lower numbers mean higher
- *   priority.
+ *   Optional. Integer between 0 and (priorityRange - 1). Specifies the
+ *   priority of the caller if there are no available resources. Lower numbers
+ *   mean higher priority.
+ *
+ * @param {Object} options
+ *   Optional. If `options.block` is `false`, and no space is available in the
+ *   queue, immediately return Pool.full (an Error object). If `options.block`
+ *   is true and `options.timeout` is set to a value greater than 0, attempt
+ *   to enqueue the item for `options.timeout` ms. If the item has not been
+ *   dequeued by that time, hit the callback with Pool.full, and don't run the
+ *   callback.
  *
  * @returns {boolean} `true` if the pool is not fully utilized, `false` otherwise.
  */
-Pool.prototype.acquire = function acquire (callback, priority) {
+Pool.prototype.acquire = function acquire (callback, priority, options) {
   if (this._draining) {
     throw new Error('pool is draining and cannot accept work')
   }
+  var that = this
+  if (options !== null && typeof options !== "undefined") {
+    if (options.block === false) {
+      if (this._isFull()) {
+        callback(fullError)
+        return false
+      }
+    } else {
+      if (typeof options.timeout === "number") {
```

Comment:
Is it better to name this option as `acquireTimeout` to avoid confusion, if any. We have other timeouts like `idleTimeout` etc.